src/generators/clientContextFileGenerator.ts

```diff
@@ -250,14 +250,14 @@ function writeDefaultOptions(
     ? `// Initializing default values for options
   if (!options) {
      options = {};
-   }
+  }
 
-  if (!options.userAgent) {
-     const defaultUserAgent = coreHttp.getDefaultUserAgentValue();
-     options.userAgent = \`\${packageName}/\${packageVersion} \${defaultUserAgent}\`;
-   }
+  const defaultUserAgent = coreHttp.getDefaultUserAgentValue();
+  options.userAgent = options.userAgent
+  ? \`\${options.userAgent} \${packageName}/\${packageVersion} \${defaultUserAgent}\`
+  : \`\${packageName}/\${packageVersion} \${defaultUserAgent}\`;
 
```

Comment:
NIT: You can build this bit once: `\${packageName}/\${packageVersion} \${defaultUserAgent}\` before the check.

Reply:
Done